GlobalstatsIO.cs

```diff
@@ -5,318 +5,315 @@
+                update = true;
+            }
+            else
+            {
+                if (name == "")
+                    name = "anonymous";
+            }
+
+            string json_payload;
+
+            if (update == false)
+            {
+                json_payload = "{\"name\":\"" + name + "\", \"values\":{";
+            }
+            else
+            {
+                json_payload = "{\"values\":{";
+            }
+
+            bool semicolon = false;
+            foreach (KeyValuePair<string, string> value in values)
+            {
+                if (semicolon)
+                    json_payload += ",";
+                json_payload += "\"" + value.Key + "\":\"" + value.Value + "\"";
+                semicolon = true;
+            }
+            json_payload += "}}";
+
+            byte[] pData = Encoding.UTF8.GetBytes(json_payload);
+            UnityWebRequest www;
+
+            if (update == false)
+            {
+                www = new UnityWebRequest(url, "POST")
+                {
+                    uploadHandler = new UploadHandlerRaw(pData)
+                };
+            }
+            else
+                www = UnityWebRequest.Put(url, pData);
+
+            www.downloadHandler = new DownloadHandlerBuffer();
+            www.SetRequestHeader("Authorization", "Bearer " + api_access_token.access_token);
+            www.SetRequestHeader("Content-Type", "application/json");
+            yield return www.SendWebRequest();
+
+            string responseBody = www.downloadHandler.text;
+
+            StatisticResponse statistic = null;
+
+            if (www.isNetworkError || www.isHttpError)
+            {
+                Debug.LogWarning("Error submitting statistic: " + www.error);
+                Debug.Log("GlobalstatsIO API Response: " + responseBody);
+                callback(false);
+            }
+            else
+            {
+                statistic = JsonUtility.FromJson<StatisticResponse>(responseBody);
+            }
+
+            // ID is available only on create, not on update, so do not overwrite it
+            if (statistic._id != null && statistic._id != "")
+                statistic_id = statistic._id;
+
+            user_name = statistic.name;
+
+            //Store the returned data statically
+            foreach (StatisticValues value in statistic.values)
+            {
+                bool updated_existing = false;
+                for (int i = 0; i < statistic_values.Count; i++)
+                {
+                    if (statistic_values[i].key == value.key)
+                    {
+                        statistic_values[i] = value;
+                        updated_existing = true;
+                        break;
+                    }
+                }
+                if (!updated_existing)
+                {
+                    statistic_values.Add(value);
+                }
+            }
+
+            callback(true);
```

Comment:
You've made callback optional: Action<bool> callback = null.
You can use: callback?.Invoke(true); so that it doesn't throw when callback == null.